add support for patch 2.4 runewords

walkawayy opened the pull request:
This adds support for patch 2.4 runewords. Here's a screenshot of how it looks:
![image](https://user-images.githubusercontent.com/81546780/166519644-c7a9181f-69b0-4adf-a89e-66e33b5f05cf.png)

I had to expand the table size to accommodate Insight which now can be made in missile weapons.

Michaelangel007 commented:
Looking at this PR today via `git fetch origin pull/38/head:pr38` ..

walkawayy commented:
Thanks for checking! I noticed two more issues:
1. All the pre patch 2.4 colors are alternating red instead of their correct colors. The 2.4 runewords show up as the yellow color I set.
2. I need to push all Rune 4s back since Mist has a Thul in slot 4.
3. Clicking rune 2 column sorting doesn't work. Instead, clicking the rune 3 column sorts rune 2 column, clicking the rune 4 column sorts rune 3 column, etc. This then breaks sort by level which sorts by name.
Sorry about that. I'll make changes for problem 2, but I may need help on problems 1 and 3.

Michaelangel007 commented:
There are 2 more issues I want to resolve first.

1. From the Javascript console you'll want to make sure `verify()` works.  For my local branch I fixed `COLUMNS` to be this:
```
//              S Tier
//                 Name
//                    Sockets
var COLUMNS = [ 0, 2, 22, 25, 57, 97, 101 ]; // Runeword Recipe Columns
//                        Type
//                            Rune#1
//                                CLevel
//                                    Version
/*
 Insight             4  Polearms/Staves/Missile Weapons Ral   + Tir   + Tal  + Sol              27  1.10 lad",
                     ^  ^                               ^                                       ^   ^
*/
```

2. It looks like the previous version has a manifest bug and is showing `1x Ral, 1x Sol, 1x Tal, 1x Tir` for the Rune manifest even though I have no runewords marked as favorites.

Michaelangel007 commented:
>  1. All the pre patch 2.4 colors are alternating red instead of their correct colors (red, green, blue). 

Looks fine on my end when I click on the `Version` button.  Screenshot?

Michaelangel007 commented:
> Clicking rune 2 column sorting doesn't work. 

Use the updated `COLUMNS` listed above which fixes this.

walkawayy commented on index.html:
```diff
@@ -2809,11 +2828,13 @@
              if( is111 ) _addClasses( elem, isOdd ? 'k11' : 'k22' );
         else if( isLad ) _addClasses( elem, isOdd ? 'b11' : 'b22' );
         else if( is110 ) _addClasses( elem, isOdd ? 'g11' : 'g22' );
+        else if( is24  ) _addClasses( elem, isOdd ? 'y11' : 'y22' );
```
Also here.

walkawayy commented on index.html:
```diff
@@ -2809,11 +2828,13 @@
         var is110 = ver.indexOf( '1.10'     ) === 0; // g
         var isLad = ver.indexOf( '1.10 lad' ) === 0; // b
         var is111 = ver.indexOf( '1.11'     ) === 0; // k
+        var is24  = ver.indexOf( '2.4  lad' ) === 0; // y
```
I added in this to try and fix the grouping and color.

walkawayy reviewed (commented):
Awesome thanks. The sorting is fixed now. And also my colors all seem to be OK now except now it's the 2.4 runewords color that isn't working. Here's a screenshot:
![image](https://user-images.githubusercontent.com/81546780/167320978-5e9c95a3-dad6-461c-9313-a5f4e29da44f.png)

I added in 2.4 to the `updateRuneWordsOrder` function but that didn't seem to work.

Michaelangel007 commented:
I'll try to take a look at this in the next day or two.

walkawayy commented:
The 2.4 runewords were showing up as the red colors (r11 and r22) which was the same color as the 1.09 runewords. I think I figured out why it was broken. I had updated the `updateRuneWordsOrder` function, but I had a typo. There was an extra space in the `'2.4 lad' index.
Bugged: `var is24  = ver.indexOf( '2.4  lad'  ) === 0; // y`
Fixed: `var is24  = ver.indexOf( '2.4 lad'  ) === 0; // y`

The 2.4 colors are working for me now:
![image](https://user-images.githubusercontent.com/81546780/168135185-53f3d868-0b70-472d-9e91-687cf1215ac4.png)

Michaelangel007 commented:
Looks good.  Ship it!

Michaelangel007 commented:
Thanks again for this patch and your patience! (I had been too lazy to test out PRs locally.  Kind of stupid really when all it was required two git commands. LUL.)

I added a special thanks to you and Wyrda65 since you were both instrumental in the updates for Patch 2.4.
